addon/components/new-bourbon-select-field.js

```diff
@@ -119,32 +124,28 @@ export default Component.extend(ClickHandlerMixin, {
 
       let index = this.get('internalContent').indexOf(valueHolder);
 
+      if ((valueHolder && !valueHolder.enabled) || ((!this.get('prompt') && index === -1))) {
+        // if value passed in that is not enabled need to check for the first enabled option
+        index = this.get('internalContent').findIndex(element => element.enabled === true);
+      }
+
       if (index !== -1) {
         return index;
-      } else if (!this.get('prompt')) {
-        // default to first option when there is no prompt passed
-        if (!this.get('defaultSelection')) {
-          this.set(
-            'value',
-            this.get('internalContent')
-              .objectAt(0)
-              .get('value')
-          );
-        }
-        return 0;
       } else {
         // if index = -1 and this.get('prompt') then show prompt
         return null;
       }
     },
 
     set(key, value) {
-      this.set(
-        'value',
-        this.get('internalContent')
-          .objectAt(value)
-          .get('value')
-      );
+      if (value !== -1) {
```

Comment:
when would this property `selectedIndex` be set to `-1`?

Reply:
when there is a prompt passed and there is no value.